Sources/SmokeAWSCredentials/ExpiringCredentials.swift

```diff
@@ -82,6 +84,23 @@ public struct ExpiringCredentials: Codable, SmokeAWSCore.Credentials {
     static func getCurrentCredentials(dataRetriever: () throws -> Data) throws -> ExpiringCredentials {
         let data = try dataRetriever()
         
-        return try jsonDecoder.decode(ExpiringCredentials.self, from: data)
+        let expiringCredentials = try jsonDecoder.decode(ExpiringCredentials.self, from: data)
+        
+        // ensure we are not getting junk credentials data
+        guard expiringCredentials.accessKeyId != nullString,
+            expiringCredentials.secretAccessKey != nullString,
+            expiringCredentials.sessionToken != nullString else {
+                let dataString = String(data: data, encoding: .utf8) ?? ""
+                
+                let reason = "Invalid credentials received: " + dataString
+                throw SmokeAWSCredentialsError.missingCredentials(reason: reason)
+        }
+        
+        if let expiration = expiringCredentials.expiration, expiration.timeIntervalSinceNow < 0 {
```

Comment:
It seems that `expiration.timeIntervalSinceNow <= 0` would be better.  I think `0` means Date.now() is the exact expiration time.

Reply:
I think that makes sense. I'll make that change.